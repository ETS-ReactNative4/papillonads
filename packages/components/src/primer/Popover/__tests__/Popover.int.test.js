import {
  renderCustom,
  renderDefault,
  renderDefaultLarge,
  renderBottom,
  renderBottomRight,
  renderBottomLeft,
  renderLeft,
  renderLeftBottom,
  renderLeftTop,
  renderRight,
  renderRightBottom,
  renderRightTop,
  renderTopLeft,
  renderTopRight,
} from './Popover.int.render'

describe('<Popover />', () => {
  const introOnClickMockFn = jest.fn()
  const acknowledgeOnClickMockFn = jest.fn()

  describe('Event', () => {
    test('must call introOnClick()', () => {
      const mountCustomRender = global.renderMount(renderCustom(introOnClickMockFn, acknowledgeOnClickMockFn))
      mountCustomRender.find('button').first().simulate('click')
      expect(mountCustomRender.props().introOnClick).toBe(introOnClickMockFn)
      expect(mountCustomRender.props().introOnClick).toHaveBeenCalled()
    })

    test('must call acknowledgeOnClick()', () => {
      const mountCustomRender = global.renderMount(renderCustom(introOnClickMockFn, acknowledgeOnClickMockFn))
      mountCustomRender.find('button').last().simulate('click')
      expect(mountCustomRender.props().acknowledgeOnClick).toBe(acknowledgeOnClickMockFn)
      expect(mountCustomRender.props().acknowledgeOnClick).toHaveBeenCalled()
    })
  })

  describe('Render', () => {
    test('must match renderDefault()', () => {
      expect(global.renderToJSON(renderDefault())).toMatchSnapshot()
    })

    test('must match renderDefaultLarge()', () => {
      expect(global.renderToJSON(renderDefaultLarge())).toMatchSnapshot()
    })

    test('must match renderBottom()', () => {
      expect(global.renderToJSON(renderBottom())).toMatchSnapshot()
    })

    test('must match renderBottomRight()', () => {
      expect(global.renderToJSON(renderBottomRight())).toMatchSnapshot()
    })

    test('must match renderBottomLeft()', () => {
      expect(global.renderToJSON(renderBottomLeft())).toMatchSnapshot()
    })

    test('must match renderLeft()', () => {
      expect(global.renderToJSON(renderLeft())).toMatchSnapshot()
    })

    test('must match renderLeftBottom()', () => {
      expect(global.renderToJSON(renderLeftBottom())).toMatchSnapshot()
    })

    test('must match renderLeftTop()', () => {
      expect(global.renderToJSON(renderLeftTop())).toMatchSnapshot()
    })

    test('must match renderRight()', () => {
      expect(global.renderToJSON(renderRight())).toMatchSnapshot()
    })

    test('must match renderRightBottom()', () => {
      expect(global.renderToJSON(renderRightBottom())).toMatchSnapshot()
    })

    test('must match renderRightTop()', () => {
      expect(global.renderToJSON(renderRightTop())).toMatchSnapshot()
    })

    test('must match renderTopLeft()', () => {
      expect(global.renderToJSON(renderTopLeft())).toMatchSnapshot()
    })

    test('must match renderTopRight()', () => {
      expect(global.renderToJSON(renderTopRight())).toMatchSnapshot()
    })
  })
})
