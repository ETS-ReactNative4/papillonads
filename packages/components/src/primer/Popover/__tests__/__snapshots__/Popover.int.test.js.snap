// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Popover /> Render must match renderBottom() 1`] = `
<div
  className="position-relative text-center"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--bottom mx-auto mb-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
</div>
`;

exports[`<Popover /> Render must match renderBottomLeft() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": true,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--bottom-left mb-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
</div>
`;

exports[`<Popover /> Render must match renderBottomRight() 1`] = `
<div
  className="position-relative text-right pr-2"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--bottom-right mb-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
</div>
`;

exports[`<Popover /> Render must match renderDefault() 1`] = `
<div
  className="position-relative text-center"
>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": true,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large mx-auto mt-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderDefaultLarge() 1`] = `
<div
  className="position-relative text-center"
>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": true,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large mx-auto mt-2 Popover-message--large"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderLeft() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--left"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderLeftBottom() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--left-bottom"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderLeftTop() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--left-top"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderRight() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--right"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderRightBottom() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--right-bottom"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderRightTop() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--right-top"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderTopLeft() 1`] = `
<div
  className="position-relative"
>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": false,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large Popover-message--top-left"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;

exports[`<Popover /> Render must match renderTopRight() 1`] = `
<div
  className="position-relative text-center"
>
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="button"
  >
    UI
  </button>
  <div
    className={
      Object {
        "pl-2": false,
        "right-0 left-0": true,
      }
    }
  >
    <div
      className="Popover-message text-left p-4 Box box-shadow-large mx-auto mt-2"
    >
      <h4
        className="mb-2"
      >
        Popover heading
      </h4>
      <p>
        Message about this particular piece of UI.
      </p>
      <button
        className="mt-2 text-bold btn btn-outline"
        onClick={[Function]}
        type="button"
      >
        Got it!
      </button>
    </div>
  </div>
</div>
`;
